cypress-io/cypress
# fix: restore spec filter in client side store

- Closes #24936

### User facing changelog


The spec search term will no longer become stale in the UI.

### Additional details


This restores some old code to keep this value up to date on the front end, and unskips a previously skipped test for this behavior.

### Steps to test


- Open Cypress at the specs list and enter a search term. 
- Open a spec from your search results and see that your latest search term is populated in the inline specs list on the spec runner page. 
- Edit the search term then navigate back to the specs list, see that the edited term is displayed.
- Quit and re-open the browser via the launchpad (as described in #24936) and see that the latest search term is correct when the browser re-opens.

### How has the user experience changed?


### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
